File: py/DS/BST.py
root = None
class Node:
    def __init__(self,x=None):
        self.data = x
        self.left = None
        self.right = None

def Create():
    n = int(input("size"))
    root = Node()
    root.data = int(input("data"))
    while(n>1):
        temp = Node()
        temp.data = int(input("data"))
        tr = root
        while(True):
            if(temp.data<tr.data):
                if(tr.left is None):
                    tr.left = temp
                    break
                else:
                    tr = tr.left
            else:
                if(tr.right is None):
                    tr.right = temp
                    break
                else:
                    tr = tr.right
        n-=1
    return root

def Pre(root):
    if(root is None):
        return
    print(root.data,end=" ")
    Pre(root.left)
    Pre(root.right)
BST = Create()
Pre(BST)

File: c++/Data_Struct/BST__.cpp
#include<iostream>
using namespace std;
struct Node
{
    int data;
    Node *left;
    Node *right;
};
Node *root = NULL;
void create(){
    int size;
    Node*temp ,*trav;
    cout<<"Enter the size of the tree: ";
    cin>>size;
    for(int i=0;i<size;i++){
        temp = new Node;
        cout<<"Enter the data: ";
        cin>>temp->data;
        temp->left = NULL;
        temp->right = NULL;
        if(root == NULL){
            root = temp;
        }
        else{
            trav = root;
            while(1){
                if(temp->data < trav->data){
                    if(trav->left == NULL){
                        trav->left = temp;
                        break;
                    }
                    else{
                        trav = trav->left;
                    }
                }
                else{
                    if(trav->right == NULL){
                        trav->right = temp;
                        break;
                    }
                    else{
                        trav = trav->right;
                    }
                }
            }
        }
    }
}
void printPostorder(Node* node) 
{ 
    if (node == NULL) 
        return; 
    printPostorder(node->left);
    printPostorder(node->right);
    cout << node->data << " "; 
} 
  
    int main() 
{  
    create();
    cout << "\nPostorder traversal of binary tree is \n"; 
    printPostorder(root); 
  
    return 0;
  
} 